browser/js/calculator.js

```diff
@@ -22,229 +28,247 @@
-    if ( data.memory === '0' ) {
-      data.current = ( parseFloat( data.current ) / 100 ).toString()
-      setOutput()
+  Calculator.prototype.convertPercent = function () {
+    if ( this.data.memory === '0' ) {
+      this.data.current = ( parseFloat( this.data.current ) / 100 ).toString()
+      this.setOutput()
     } else {
-      data.current = (
-        (parseFloat( data.current ) / parseFloat( data.memory )
+      this.data.current = (
+        (parseFloat( this.data.current ) / parseFloat( this.data.memory )
       ) * 100 ).toString()
-      setOutput()
+      this.setOutput()
     }
   }
-  function addDot() {
-    if ( data.current.indexOf( '.' ) === -1 ) {
-      data.current = data.current + '.'
-      setOutput()
+  Calculator.prototype.addDot = function () {
+    if ( this.data.current.indexOf( '.' ) === -1 ) {
+      this.data.current = this.data.current + '.'
+      this.setOutput()
     }
   }
 
-  function calculate() {
-    if ( !(data.memory === '0' && data.current === '0') ) {
-      switch( data.operation ) {
-        case buttons.ADD:
-          add()
+  Calculator.prototype.calculate = function () {
+    if ( !(this.data.memory === '0' && this.data.current === '0') ) {
+      switch( this.data.operation ) {
+        case Calculator.buttons.ADD:
+          this.add()
           break;
-        case buttons.SUBTRACT:
-          subtract()
+        case Calculator.buttons.SUBTRACT:
+          this.subtract()
           break;
-        case buttons.MULTIPLY:
-          multiply()
+        case Calculator.buttons.MULTIPLY:
+          this.multiply()
           break;
-        case buttons.DIVIDE:
-          divide()
+        case Calculator.buttons.DIVIDE:
+          this.divide()
           break;
       }
-      setOutput( data.result )
+      this.setOutput( this.data.result )
     }
   }
-  function add() {
-    setResult( parseFloat( data.memory ) + parseFloat( data.current ) )
+  Calculator.prototype.add = function () {
+    this.setResult(
+      parseFloat( this.data.memory ) + parseFloat( this.data.current ) )
   }
-  function subtract() {
-    setResult( parseFloat( data.memory ) - parseFloat( data.current ) )
+  Calculator.prototype.subtract = function () {
+    this.setResult(
+      parseFloat( this.data.memory ) - parseFloat( this.data.current ) )
   }
-  function multiply() {
-    setResult( parseFloat( data.memory ) * parseFloat( data.current ) )
+  Calculator.prototype.multiply = function () {
+    this.setResult(
+      parseFloat( this.data.memory ) * parseFloat( this.data.current ) )
   }
-  function divide() {
-    setResult( parseFloat( data.memory ) / parseFloat( data.current ) )
+  Calculator.prototype.divide = function () {
+    this.setResult(
+      parseFloat( this.data.memory ) / parseFloat( this.data.current ) )
   }
+
+  document.querySelectorAll( '.calculator' ).forEach( function( calculator ) {
+    var calc = new Calculator( calculator )
+    calc.addListeners()
```

Comment:
You don't need to call `.addListeners()` here. Just do this work in the `Constructor` function.